scim/usersRecords.bal

```diff
@@ -0,0 +1,454 @@
+# + emails - Email addresses for the User 
+# + addresses - Physical mailing address for the user  
+# + phoneNumbers - Phone numbers for the user  
+# + ims - Instant messaging address for the user  
+# + roles - List of roles for the user  
+# + photos - URI of image of the user  
+# + userType - Relationship between the organization and the user  
+# + title - Title of the user
+# + preferredLanguage - Preferred written or spoken languages of the user
+# + locale - Default location of the user
+# + timezone - Time zone of the user
+# + active - Administrative status of the user 
+# + password - Password of the user
+# + groups - Groups to which the user belongs  
+# + x509Certificates - Certificates associated with the user 
+# + meta - Metadata of the user  
+# + urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User - Attributes belong to business or enterprise extension schema
+# + urn\:scim\:wso2\:schema - Attributes belong to WSO2 extension schema
+public type UserCreate record {
+    string[]|string schemas?;
+    int externalId?;
+    string userName?;
+    Name name?;
+    string displayName?;
+    string nickName?;
+    string profileUrl?;
+    Email[] emails?;
+    Address[] addresses?;
+    Phone[] phoneNumbers?;
+    Ims[] ims?;
+    Role[] roles?;
+    Photo[] photos?;
+    string userType?;
+    string title?;
+    string preferredLanguage?;
+    string locale?;
+    string timezone?;
+    boolean active?;
+    string password;
+    Group[] groups?;
+    Certificate[] x509Certificates?;
+    Meta meta?;
+    SCIMEnterpriseUser urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User?;
+    Custom urn\:scim\:wso2\:schema?;
+};
+
+# Represents the SCIM Enterprise User extensioon resource.
+#
+# + employeeNumber - Identifier in association with the organization.  
+# + costCenter - Name of a cost center  
+# + organization - Name of an organization  
+# + division - Name of a division  
+# + department - Name of a department  
+# + manager - Manager of the user  
+# + verifyEmail - Email address has been verified or not
+public type SCIMEnterpriseUser record {
+    string employeeNumber?;
+    int costCenter?;
+    string organization?;
+    string division?;
+    string department?;
+    Manager manager?;
+    boolean verifyEmail?;
+};
+
+# Represents the sub-attributes of the manager attribute in the SCIMEnterpriseUser record.
+#
+# + value - Id of the SCIM resource representing the manager of the user
+# + displayName - DisplayName of the manager of the user  
+# + \$ref - Reference URI 
+public type Manager record {
+    string value?;
+    string displayName?;
+    string \$ref?;
+};
+
+# Represents the sub-attributes of the emails complex attribute in SCIMUser record.
+#
+# + value - Canonicalized representation of the email value  
+# + 'type - Email classification  
+# + primary - Whether the primary email address
+public type Email record {
+    string value?;
+    string 'type?;
```

Comment:
can we define the cannonical values of the type using ENUM or any other form
https://www.rfc-editor.org/rfc/rfc7643.html#:~:text=work%27%20or%20%27home%27.%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22required%22%20%3A%20false%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22caseExact%22%20%3A%20false%2C-,%22canonicalValues%22%20%3A%20%5B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%22work%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%22home%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%22other%22%0A%20%20%20%20%20%20%20%20%20%20%20%20%5D,-%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22mutability%22%20%3A%20%22readWrite%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22returned%22%20%3A%20%22default%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22uniqueness%22%20%3A%20%22none%22%0A%20%20%20%20%20%20%20%20%20%20%7D%2C%0A%20%20%20%20%20%20%20%20%20%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%22name

Reply:
The same comment applies to other canonical values defined in the spec.
eg: types of ims, types of address, types of phone numbers